src/components/esphome-button-menu.ts

```diff
@@ -0,0 +1,50 @@
+import "@material/mwc-menu";
+import type { Corner, Menu } from "@material/mwc-menu";
+import { css, CSSResultGroup, html, LitElement, TemplateResult } from "lit";
+import { customElement, property, query } from "lit/decorators.js";
+
+@customElement("esphome-button-menu")
+export class ESPHomeButtonMenu extends LitElement {
+  @property() public corner: Corner = "TOP_START";
+
+  @query("mwc-menu", true) private _menu?: Menu;
+
+  public get items() {
+    return this._menu?.items;
+  }
+
+  public get selected() {
+    return this._menu?.selected;
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <div @click=${this._handleClick}>
```

Comment:
We should use `<button>` for buttons

Reply:
Hmm, then we'd have to remove all the styling from the default `<button>` - or we make the component directly use `<mwc-icon-button>` and pass in the icon (which the slot is filled with in the parent)

Reply:
ok we can leave it like this. For Improv we actually add the click listener to the slot itself. https://github.com/improv-wifi/sdk-js/blob/main/src/launch-button.ts#L75-L79